src/action.ts

```diff
@@ -57,6 +57,7 @@ class Action {
 
     core.startGroup("Uploading artifacts...")
     this.uploadArtifacts(actionResult.baseDir, actionResult.artifacts, executionGraph.execution_graph_id)
+    this.rmdir(actionResult.baseDir)
```

Comment:
Couple of points here:
* The latest PR was causing failures because we're not awaiting for the upload to complete, therefore, we were removing the directory before the artifacts were pushed. **We should `await` for `this.uploadArtifacts`**
* We should await for `rmdir` as well (see my other comment)